src/CaptureServiceBase/include/CaptureServiceBase/UploaderClientCaptureEventCollectorBuilder.h

```diff
@@ -5,14 +5,29 @@
 #ifndef CAPTURE_SERVICE_BASE_UPLOADER_CLIENT_CAPTURE_EVENT_COLLECTOR_BUILDER_H_
 #define CAPTURE_SERVICE_BASE_UPLOADER_CLIENT_CAPTURE_EVENT_COLLECTOR_BUILDER_H_
 
+#include <absl/base/thread_annotations.h>
+#include <absl/synchronization/mutex.h>
+
 #include "CaptureServiceBase/ClientCaptureEventCollectorBuilder.h"
+#include "CaptureUploader/UploadDataInterface.h"
 
 namespace orbit_capture_service_base {
 
-// Create a `ClientCaptureEventCollectorBuilder` which builds a `UploaderCaptureEventCollector` for
-// the cloud collector.
-[[nodiscard]] std::unique_ptr<ClientCaptureEventCollectorBuilder>
-CreateUploaderClientCaptureEventCollectorBuilder();
+// A `ClientCaptureEventCollectorBuilder` implementation to build
+// `UploaderClientCaptureEventCollector` and provide access to a `UploadDataInterface` for the cloud
+// collector.
+class UploaderClientCaptureEventCollectorBuilder : public ClientCaptureEventCollectorBuilder {
+ public:
+  [[nodiscard]] std::unique_ptr<orbit_producer_event_processor::ClientCaptureEventCollector>
+  BuildClientCaptureEventCollector() override;
+
+  // This method is blocked until BuildClientCaptureEventCollector is called.
+  [[nodiscard]] orbit_capture_uploader::UploadDataInterface* GetUploadDataInterface();
```

Comment:
Maybe `WaitForUploadDataInterface`?

Reply:
As now the manager (builder is renamed as manager) owns the collector and creates the collector in its constructor, we do not need to wait here any more.